app/src/main/java/org/fossasia/pslab/sensorfragment/SensorFragmentMain.java

```diff
@@ -64,35 +62,63 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         sensorAddr.put(0x40, "SHT21");
         sensorAddr.put(0x39, "TSL2561");
 
+        try {
+            data = i2c.scan(null);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if (data != null) {
+            for (Integer myInt : data) {
+                if (sensorAddr.get(myInt) != null) {
+                    dataAddress.add(String.valueOf(myInt));
+                    dataName.add(sensorAddr.get(myInt));
+                }
+            }
+            for (String s : dataAddress) {
+                tvData += s + ":" + sensorAddr.get(Integer.parseInt(s)) + "\n";
+            }
+        }
+        String[] dataDisp = dataName.toArray(new String[dataName.size()]);
+        adapter = new ArrayAdapter<>(getContext(), R.layout.sensor_list_item, R.id.tv_sensor_list_item, dataDisp);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.sensor_main, container, false);
+
         Button buttonSensorAutoscan = (Button) view.findViewById(R.id.button_sensor_autoscan);
-        final TextView tvSensorScan = (TextView) view.findViewById(R.id.tv_sensor_scan);
+        tvSensorScan = (TextView) view.findViewById(R.id.tv_sensor_scan);
+        setText(tvData);
         lvSensor = (ListView) view.findViewById(R.id.lv_sensor);
+        setAdapter(adapter);
 
         buttonSensorAutoscan.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 if (scienceLab.isConnected()) {
-                    String tvData = "";
                     try {
                         data = i2c.scan(null);
                     } catch (IOException e) {
                         e.printStackTrace();
                     }
+
                     if (data != null) {
                         for (Integer myInt : data) {
-                            if (sensorAddr.get(myInt) != null) {
+                            if (sensorAddr.get(myInt) != null && myInt != null) {
```

Comment:
If ```myInt``` is null, the current expression would first try to find ```null``` in map.
Its should be ```myInt != null && sensorAddr.get(myInt)```